cmd/snap/cmd_run.go

```diff
@@ -504,6 +524,14 @@ func (x *cmdRun) snapRunApp(snapApp string, args []string) error {
 		}
 	}
 
+	if hint != runinhibit.HintNotInhibited {
+		// Get "current" snap info assuming snap run inhibition is removed
+		info, app, err = getInfoAndApp(snapName, appName, snap.R(0))
+		if err != nil {
+			return err
+		}
```

Comment:
can this race with the snap becoming inhibited again? do we need to ask this while we hold the lock ourselves for a short time?

Reply:
great catch, thank you! it seems to be rare that this could happen but definitely possible. I am assuming by locking you mean getting a read lock on the hint file and not the snap lock, right? I updated the PR with this assumption, please let me know If this is not what you meant.

Reply:
to be clear my worry here was that we would end up inhibited again which means current symlink might be missing again and then we would go back to the user with a perplexing error